test/src/main/java/org/apache/accumulo/test/functional/ThriftMaxFrameSizeIT.java

```diff
@@ -56,50 +62,95 @@ class TestDefault extends TestMaxFrameSize {
     TestDefault() {
       serverType = ThriftServerType.getDefault();
     }
+
+    @Test
+    public void testDefaultServerFrameSize() throws Exception {
+      AtomicBoolean succeeded = new AtomicBoolean(false);
+      assertThrows(AssertionFailedError.class,
+          () -> assertTimeoutPreemptively(Duration.ofSeconds(30),
+              () -> testMaxFrameSizeLargerThanDefault(succeeded)));
+      assertFalse(succeeded.get());
+    }
   }
 
   @Nested
   class TestThreadedSelector extends TestMaxFrameSize {
     TestThreadedSelector() {
       serverType = ThriftServerType.THREADED_SELECTOR;
     }
+
+    @Test
+    public void testThreadedSelectorServerFrameSize() throws Exception {
+      AtomicBoolean succeeded = new AtomicBoolean(false);
+      assertThrows(AssertionFailedError.class,
+          () -> assertTimeoutPreemptively(Duration.ofSeconds(30),
+              () -> testMaxFrameSizeLargerThanDefault(succeeded)));
+      assertFalse(succeeded.get());
+    }
   }
 
   @Nested
   class TestCustomHsHa extends TestMaxFrameSize {
     TestCustomHsHa() {
       serverType = ThriftServerType.CUSTOM_HS_HA;
     }
+
+    @Test
+    public void testCustomHsHaServerFrameSize() throws Exception {
+      AtomicBoolean succeeded = new AtomicBoolean(false);
+      assertThrows(AssertionFailedError.class,
+          () -> assertTimeoutPreemptively(Duration.ofSeconds(30),
+              () -> testMaxFrameSizeLargerThanDefault(succeeded)));
+      assertFalse(succeeded.get());
+    }
   }
 
   @Nested
   class TestThreadPool extends TestMaxFrameSize {
     TestThreadPool() {
       serverType = ThriftServerType.THREADPOOL;
     }
+
+    @Test
+    public void testThreadPoolServerFrameSize() throws Exception {
+      AtomicBoolean succeeded = new AtomicBoolean(false);
+      assertThrows(AssertionFailedError.class,
+          () -> assertTimeoutPreemptively(Duration.ofSeconds(30),
+              () -> testMaxFrameSizeLargerThanDefault(succeeded)));
+      assertFalse(succeeded.get());
+    }
   }
 
   @Nested
   class TestSsl extends TestMaxFrameSize {
     TestSsl() {
-      serverType = ThriftServerType.THREADPOOL;
+      serverType = ThriftServerType.SSL;
+    }
+
+    @Test
+    public void testSslServerFrameSize() throws Exception {
+      AtomicBoolean succeeded = new AtomicBoolean(false);
+      testMaxFrameSizeLargerThanDefault(succeeded);
+      assertTrue(succeeded.get());
```

Comment:
I'm really not sure why this is still working. I guess Ssl transports don't care about the max message size for some reason? That's weird.

Reply:
After refactoring the test, I don't see the ssl case being an exception... though it does seem to fail with slightly different messages.